- Adding support for GradientBoostClassification/Regression

harshitpandey091 opened the pull request:
- Correcting the continuousSplit logic in DecisionTreeModelInfo.
- Taking utility/src classes from 1.6 -> 2.0 adapters
- Adding support for GradientBoostClassification/Regression

Commit:
- Adding support for GradientBoostClassification/Regression
- Correcting the continuousSplit logic in DecisionTreeModelInfo.
- Taking utility/src classes from 1.6 -> 2.0 adapters

Commit:
Compilation fix : Adding iscontinuousSplit (at tree level) for backward
compatibility

Commit:
Adding changes to 1.6

Commit:
- Adding testcases for GBT/Decision trees

akshay-sharma commented on adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeClassificationModelBridgeTest.java:
```diff
@@ -0,0 +1,57 @@
+
+
+    @Test
+    public void testDecisionTreeRegressionPrediction() {
```
Please add a test using this in pipeline as well. You can refer 1.6 test for the same.

akshay-sharma commented on adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeRegressionModelBridgeTest.java:
```diff
@@ -0,0 +1,60 @@
+ * Created by akshay.us on 8/29/16.
+ */
+public class DecisionTreeRegressionModelBridgeTest extends SparkTestBase {
+
```
Please add a test using this in a pipeline as well. You can refer 1.6 test for the same.

akshay-sharma commented on adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapterTest.java:
```diff
@@ -0,0 +1,58 @@
+ * Created by akshay.us on 8/29/16.
+ */
+public class GradientBoostClassificationModelInfoAdapterTest extends SparkTestBase {
+
```
Please add a test using this in a pipeline as well.

akshay-sharma commented on adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java:
```diff
@@ -0,0 +1,65 @@
+    private static final DecisionTreeRegressionModelInfoAdapter DECISION_TREE_ADAPTER = new DecisionTreeRegressionModelInfoAdapter();
+    
+    @Override
+    GradientBoostModelInfo getModelInfo(final GBTClassificationModel sparkRfModel) {
```
Please rename variable sparkRFmodel to something related to GBmodel

akshay-sharma commented on models-info/src/main/java/com/flipkart/fdp/ml/transformer/GradientBoostClassificationTransformer.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public void transform(Map<String, Object> input) {
+        double[] inp = (double[]) input.get(forest.getInputKeys().iterator().next());
+        input.put(forest.getOutputKeys().iterator().next(), predict(inp));
```
please add raw prediction as well as class prediction as output.

harshitpandey091 replied:
GradientBoost doesn't emits rawPrediction as of now.
https://spark.apache.org/docs/latest/ml-classification-regression.html#gradient-boosted-trees-gbts

akshay-sharma reviewed (changes requested):
Overall, the changes look good to me. Please incorporate the said suggestions for completeness.

Commit:
- Added pipeline testcases
- Added checks for matching rawPrediction (only in case of decision tree
classification)

Commit:
- Adding more testcases (pipeline, rawprediction)
- minor changes on description/variable names

harshitpandey091 commented on adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java:
```diff
@@ -0,0 +1,65 @@
+package com.flipkart.fdp.ml.adapter;
```
Will take this up later in another review. Will take sometime.

harshitpandey091 commented:
Added pipeline test-cases and rawPrediction checks.
Regarding GBT in 1.6, There are a few differences between 1.6 and 2.0, which will take sometime for me to completely understand. Can take that up as a separate commit in the future

akshay-sharma reviewed (approved):
LGTM
